leader.go

```diff
@@ -81,22 +81,22 @@ func (k *Kinsumer) becomeLeader() {
 			}
 		}
 	}()
-	k.isLeader = true
+
+	atomic.StoreInt32(&k.leader, 1)
 }
 
 // unbecomeLeader stops the leadership goroutine.
 func (k *Kinsumer) unbecomeLeader() {
-	if !k.isLeader {
+	if !k.isLeader() {
 		return
 	}
-	if k.leaderLost == nil {
-		log.Printf("Lost leadership but k.leaderLost was nil")
-	} else {
-		close(k.leaderLost)
```

Comment:
Closing the channel causes readers to proceed, could removing this part cause unintended effects?

Reply:
I don't think so. The only reader is a goroutine in `becomeLeader` which has the same behaviour with `close(k.leaderLost)` and `k.leaderLost <- struct{}{}`.